Chemistry/tests/SamplesTests/DocsSecondQuantization.cs

```diff
@@ -0,0 +1,340 @@
+            // a tuple.
+            var ladderOperator1 = new LadderOperator<int>((creationEnum, spinOrbitalInteger));
+
+            Assert.True(ladderOperator0 == ladderOperator1);
+        }
+
+        [Fact]
+        static void FermionTerms()
+        {
+            // Let us initialize an array of tuples representing the
+            // desired sequence of creation operators.
+            var indices = new[] { (RaisingLowering.u, 1), (RaisingLowering.u, 2) };
+
+            // We can convert this array of tuples to a sequence of ladder
+            // operators using the `ToLadderSequence()` methods.
+            var ladderSequences = indices.ToLadderSequence();
+
+            // Sequences of ladder operators are quite general. For instance,
+            // they could be bosonic operators, intead of fermionic operators.
+            // We specialize them by constructing a `FermionTerm` representing 
+            // a fermion creation operator on the index `2` followed by `1`.
+            var fermionTerm = new FermionTerm(ladderSequences);
+
+            Assert.True(fermionTerm.Sequence.SequenceEqual(ladderSequences.Sequence));
+            Assert.True(fermionTerm.Coefficient == 1);
+        }
+
+        [Fact]
+        static void NumberOperator()
+        {
+            // Let us use a new method to compactly create a sequence of ladder
+            // operators. Note that we have ommitted specifying whether the 
+            // operators are raising or lowering. In this case, the first half
+            // will be raising operators, and the second half will be lowering 
+            // operators.
+            var indices = new[] { 1, 1 }.ToLadderSequence();
+
+            // We now construct a `FermionTerm` representing an annihilation operator
+            // on the index 1 followed by the creation operator on the index 1.
+            var fermionTerm0 = new FermionTerm(indices);
+
+            Assert.True(fermionTerm0.Sequence.SequenceEqual(indices.Sequence));
+            Assert.True(indices.Sequence[0].Type == RaisingLowering.u);
+
+            Assert.True(indices.Sequence[1].Type == RaisingLowering.d);
+        }
+
+        [Fact]
+        static void InequivalentLadderSequence()
+        {
+            // Let us initialize an array of tuples representing the
+            // desired sequence of creation operators.
+            var indices = new[] { (RaisingLowering.u, 1), (RaisingLowering.u, 2) };
+
+            // We now construct a `LadderSequence` representing a creation operator
+            // on the index 1 followed by 2, then a term with the reverse ordering.
+            var laddderSeqeunce = indices.ToLadderSequence();
+            var laddderSeqeunceReversed = indices.Reverse().ToLadderSequence();
+
+            // The following Boolean is `false`.
+            var equal = laddderSeqeunce == laddderSeqeunceReversed;
+
+            Assert.False(equal);
```

Comment:
To improve diagnostic information, I'd suggest asserting that the two are not equal here.
```suggestion
            Assert.NotEqual(ladderSequence, ladderSequenceReversed);
```